src/commands/register.ts

```diff
@@ -51,29 +71,30 @@ async function subcmd_update(interaction, db: Db) {
         );
 
         if (!result) {
-            interaction.reply({
+            this.interaction.reply({
             content:
                 "They found my wire... I couldn't update you; check with an admin.",
             ephemeral: true,
             });
         } else {
-            interaction.reply({
+            this.interaction.reply({
             content:
                 "Caught one; you've been updated! Feel free to update again with /register update",
             ephemeral: true,
             });
         }
+    }
 }
 
 function cmd_register(interaction, db: Db) {
     const commands = {
-        "new": subcmd_new,
-        "update": subcmd_update
+        "new": NewSubcommand,
+        "update": UpdateSubcommand
     };
 
-    // Wrap function call to pass same args to all methods
     const call_fn = commands[interaction.options.getSubcommand()];
-    call_fn(interaction, db);
+    const subcmd: MessageExecutable<CommandInteraction> = new call_fn(interaction, db);
```

Comment:
Because it's a constructor now, I renamed `call_fn` to `Action` so it would look less odd after a `new`

Reply:
Addressed in 2621dee554385cda79c1ab2d33c7350078b13e3f.